### DISCUSSION TITLE
Shortcut for subscript and superscript

### DISCUSSION THREAD

**Discussion by nema69** [TriliumNext/Trilium#3759] [Ideas] [answered]
I use trilium for taking notes in university and use subscripts a lot. It would be nice if there was a shortcut for switching the text style to subscript or superscript similar to what crtl+b does for switching to bold text. I checked the documentation of ckeditor and the trilium shortcuts and couldn't find anything. Please let me know if I'm missing anything and if this isn't implemented yet, I would love it if someone would do it :)

**Comment by zadam** [comment]
I don't think this is currently possible, sorry.

**Reply by zadam** [reply]
Update: I started experimenting with some CKEditor keyboard shortcuts and this will likely come in the near future.

**Reply by SiriusXT** [reply]
This is awesome. In addition, I hope to add a small function that can automatically expand when the mouse is hovering over the expandable button of the tool. For example, when the mouse is hovering over the table button or color button, the drop-down options can be automatically expanded. This will facilitate quick operation.
![图片](https://github.com/zadam/trilium/assets/37627919/d08b3c0c-3636-436c-805c-87fcec2b39ba)
![图片](https://github.com/zadam/trilium/assets/37627919/68a292b4-f197-452e-8a5c-157f6af42f3b)

**Comment by SiriusXT** [comment, marked as answer]
I implemented this function by using a script to simulate the mouse, the code is:
#widget
```
setTimeout(() => {
api.bindGlobalShortcut("alt+[",function (){
    if (window.getSelection().toString()!=""){
        setTimeout(() => {document.querySelectorAll("div.ck.ck-balloon-panel.ck-balloon-panel_visible.ck-balloon-panel_with-arrow.ck-toolbar-container button.ck.ck-button:nth-child(6)")[0].click();}, 10); }
    else{return}
    
}); 
api.bindGlobalShortcut("alt+]",function (){
    if (window.getSelection().toString()!=""){
        setTimeout(() => {document.querySelectorAll("div.ck.ck-balloon-panel.ck-balloon-panel_visible.ck-balloon-panel_with-arrow.ck-toolbar-container button.ck.ck-button:nth-child(7)")[0].click();}, 10); }
    else{return}
    
}); 

}, 1000); 
```
The shortcut key for superscript is alt+[, the shortcut key for subscript is alt+].

The following is the implementation of shortcut keys for inserting formulas that I use myself (It can convert the selected text into a formula, and the default shortcut key (ctrl+m) can only insert the formula):


```


setTimeout(() => {
api.bindGlobalShortcut("alt+=",function (){
    if (window.getSelection().toString()!=""){
        var clip=window.getSelection().toString();
        setTimeout(() => {
            window.getSelection().deleteFromDocument();
       }, 50); 
        setTimeout(() => {document.querySelectorAll("button.ck.ck-button.ck-off.ck-block-toolbar-button:not(.ck-hidden)")[0].click();}, 100); 
    setTimeout(() => {
        document.querySelectorAll("div.ck.ck-balloon-panel.ck-balloon-panel_visible.ck-balloon-panel_with-arrow.ck-toolbar-container button.ck:nth-child(12)")[0].click();}, 110); 
        setTimeout(() => {
            var input=document.querySelectorAll(' div.ck-balloon-rotator__content form.ck.ck-math-form div.ck-math-view div.ck.ck-labeled-input input.ck.ck-input.ck-input-text')[0];
            clip = clip.replace(/^[$]+|[$]+$/g, "");
            clip = clip.replace(/^\${2,}|\${2,}$/g, "");
            input.value=clip;
       input.dispatchEvent(new Event('input'));
        }, 200); 
    setTimeout(() => {
        document.querySelectorAll("button.ck.ck-button.ck-off.ck-button-save")[0].click();//Automatically click the save button   
    }, 250); 
     }       
    else
    {
        setTimeout(() => {document.querySelectorAll("button.ck.ck-button.ck-off.ck-block-toolbar-button:not(.ck-hidden)")[0].click();}, 10); 
        setTimeout(() => {document.querySelectorAll("div.ck.ck-balloon-panel.ck-balloon-panel_visible.ck-balloon-panel_with-arrow.ck-toolbar-container button.ck:nth-child(12)")[0].click();}, 20); 
    }
}
);       
       }, 1000); 
```
The shortcut key for inserting a formula is alt+=.

**Reply by zerebos** [reply]
For the first part of this you can also do something like this with the editor directly:
```js
api.getActiveContextTextEditor().then(editor => {
    editor.keystrokes.set("Ctrl+S", "superscript");
    editor.keystrokes.set("Ctrl+Shift+S", "subscript");
});
```
